sdk/test/UnitTests/HttpWebRequestTracingExtensionTests.cs

```diff
@@ -60,19 +62,45 @@ public void TestGetResponseTraced()
             var request = (HttpWebRequest)WebRequest.Create(URL);
 
             AWSXRayRecorder.Instance.BeginSegment("parent", TraceId);
-            request.GetResponseTraced();
-            var segment = AWSXRayRecorder.Instance.TraceContext.GetEntity();
-            AWSXRayRecorder.Instance.EndSegment();
+            using (request.GetResponseTraced())
+            {
+                var segment = AWSXRayRecorder.Instance.TraceContext.GetEntity();
+                AWSXRayRecorder.Instance.EndSegment();
+
+                Assert.IsNotNull(request.Headers[TraceHeader.HeaderKey]);
 
-            Assert.IsNotNull(request.Headers[TraceHeader.HeaderKey]);
+                var requestInfo = segment.Subsegments[0].Http["request"] as Dictionary<string, object>;
+                Assert.AreEqual(URL, requestInfo["url"]);
+                Assert.AreEqual("GET", requestInfo["method"]);
 
-            var requestInfo = segment.Subsegments[0].Http["request"] as Dictionary<string, object>;
-            Assert.AreEqual(URL, requestInfo["url"]);
-            Assert.AreEqual("GET", requestInfo["method"]);
+                var responseInfo = segment.Subsegments[0].Http["response"] as Dictionary<string, object>;
+                Assert.AreEqual(200, responseInfo["status"]);
+                Assert.IsNotNull(responseInfo["content_length"]);
+            }
+        }
+        
+        /// <summary>
+        /// Ensures that when tracing is disabled that HTTP requests can execute as normal. \
+        /// See https://github.com/aws/aws-xray-sdk-dotnet/issues/57 for more information. 
+        /// </summary>
+        [TestMethod]
+        public void TestGetResponseTraced_XrayDisabled()
+        {
+            _recorder = new MockAWSXRayRecorder() { IsTracingDisabledValue = true };
+#if NET45
+            AWSXRayRecorder.InitializeInstance(_recorder);
+#else
+            AWSXRayRecorder.InitializeInstance(recorder: _recorder);
+# endif
+            Assert.IsTrue(AWSXRayRecorder.Instance.IsTracingDisabled());
 
-            var responseInfo = segment.Subsegments[0].Http["response"] as Dictionary<string, object>;
-            Assert.AreEqual(200, responseInfo["status"]);
-            Assert.IsNotNull(responseInfo["content_length"]);
+            var request = (HttpWebRequest)WebRequest.Create(URL);
+            
+            using (var response = request.GetResponseTraced() as HttpWebResponse)
```

Comment:
can we add unit test for `GetAsyncResponseTraced `

Reply:
Pushed a fix for this.